lib/common/common/Int64Math.cpp

```diff
@@ -30,6 +37,16 @@ Int64Math::Mul(int64 left, int64 right, int64 *pResult)
     return (left != 0 && right != 0 && (*pResult / left) != right);
 #endif
 }
+#else
+// FIXME: this probably needs checks for __int128 at least.
+bool
+Int64Math::Mul(int64 left, int64 right, int64 *pResult)
+{
+    __int128 result = (__int128)left * (__int128)right;
+    *pResult = (int64)result;
+    return (left != 0 && right != 0 && (*pResult / left) != right);
```

Comment:
Since the signature of the method is duplicated from above, can you move the `#ifdef _WIN32` logic into the original method body?

Since the return statement is the same as the one in `!_M_X64` branch I think there might be a way to reuse that rather than duplicating it but I see the difficulty.